crates/router/src/db/configs.rs

```diff
@@ -86,8 +87,17 @@ impl ConfigInterface for Store {
         &self,
         key: &str,
     ) -> CustomResult<storage::Config, errors::StorageError> {
-        cache::get_or_populate_cache(self, key, || async { self.find_config_by_key(key).await })
-            .await
+        let val = CONFIG_CACHE.get_val::<storage::Config>(key);
+        if let Some(val) = val {
+            Ok(val)
+        } else {
+            let val = cache::get_or_populate_cache(self, key, || async {
+                self.find_config_by_key(key).await
+            })
+            .await?;
+            CONFIG_CACHE.push(key.to_string(), val.clone()).await;
+            Ok(val)
+        }
```

Comment:
It would be preferable if `get_or_populate_cache()` either handled three levels of hierarchy, or if the function was named `get_or_populate_redis()`.

Personally, I'd prefer the former.